### DISCUSSION TITLE
Triggering onChange event manually

### DISCUSSION THREAD

**Discussion by minimaluminium** [react-hook-form/react-hook-form#2543] [Q&A]
Hello everyone!

[Here is](https://codesandbox.io/s/clever-darwin-5gl5e) a simplified version of what I'm trying to do. In this example:

* There is an editor which is only a `textarea` field. The editor height is dynamic and it's calculated on `onInput` event.
* The editor's value is fetched from the server, but has been simulated with `setTimeout()` in this example.
* After fetching from the server, I used `setValue` to set the initial value.

The problem is that `setValue` doesn't trigger `onChange` event, so the editor height is not calculated when its value is fetched from the server. One option is [sharing the ref usage](https://react-hook-form.com/faqs#Howtosharerefusage) and calculating the height in the parent component (index page in this example). But it doesn't feel like a good solution.

Is there any way to trigger `onChange` event manually? Or is there any other solution which I'm missing?

**Comment by bluebill1049** [comment]
you can build it into a Controlled input and wrapped with Controller.

**Reply by bluebill1049** [reply]
actually you are already building a controlled input, just wrapped it with Controller and expose, value, onChange props.

**Reply by bluebill1049** [reply]
https://react-hook-form.com/api#Controller
